torch_xla/csrc/runtime/pjrt_computation_client.cc

```diff
@@ -704,6 +692,16 @@ PjRtComputationClient::ExecuteReplicated(
   // Required as of cl/518733871
   execute_options.use_major_to_minor_data_layout_for_callbacks = true;
 
+  // Grab the shared lock and block the `WaitDeviceOps` until buffer is
+  // ready. Since this is the SPMD code path. There is no points to grab
+  // devices lock for every individual device.
+  TF_VLOG(5) << "ExecuteReplicated acquiring PJRT device lock for "
+             << spmd_device_str;
+  auto op = operation_tracker_.StartOperation(spmd_device_str);
+  // auto lock = lock_device_shared(spmd_device_str);
+  TF_VLOG(5) << "ExecuteReplicated acquiring PJRT device lock for "
+             << spmd_device_str << " Done";
+
   std::optional<std::vector<xla::PjRtFuture<xla::Status>>> returned_futures(
```

Comment:
You probably don't need to initialize `returned_futures`, iirc -- you can simply pass in an uninitialized one (`std::vector<xla::PjRtFuture<xla::Status>> returned_futures;`) and the runtime client will do the rest.

Reply:
Good catch. I'm always happy to delete a bit of code.